# Problem Statment:1
    
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units.
A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% 
significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

Minitab File : Cutlets.mtw



H0 : mu1 == mu2 or mu1-mu2 =0 i.e No difference in the diameter of the cutlet between two units
H1 : mu1 != mu2 or mu1-mu2 !=0 i.e There is a diffrence in the diameter of the cutlet between two units
# Perform 2 tail 2 smaple ttest

In [105]:

import pandas as pd
import numpy as np
from scipy import stats

In [106]:
df_Cutlets = pd.read_csv('/Users/Akshay/Documents/DataScience/ExcelR/Assignments/DataScience/HypothesisTesting/Cutlets.csv')
df_Cutlets.head(5)

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [107]:

print('Data Type in Data Frame ---',df_Cutlets.dtypes,'\n' , '\n','Data frame Columns Name --- ',df_Cutlets.columns,'\n', 'No of rows in df ---',df_Cutlets.shape[0], '\n', 
        'No of columns in df ---',df_Cutlets.shape[1]  )

Data Type in Data Frame --- Unit A    float64
Unit B    float64
dtype: object 
 
 Data frame Columns Name ---  Index(['Unit A', 'Unit B'], dtype='object') 
 No of rows in df --- 35 
 No of columns in df --- 2


In [108]:
df_Cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [109]:
n= df_Cutlets.shape[0]
print('Size of the sample ---', n)
muA = df_Cutlets['Unit A'].mean()
muB = df_Cutlets['Unit B'].mean()
print('Mean of UnitA ---', muA, 'Mean of UnitB ---', muB)

Size of the sample --- 35
Mean of UnitA --- 7.01909142857143 Mean of UnitB --- 6.964297142857142


In [110]:
#Perform ttest as std deviation of population is not given
t_value,p_value = stats.ttest_ind(df_Cutlets['Unit A'],df_Cutlets['Unit B'])
print('t statistic value -- ',t_value,'\n','p value of both the units',p_value)

#5% significance level
alpha = 0.05

#compare p value with alpha value
p_value > alpha


t statistic value --  0.7228688704678061 
 p value of both the units 0.4722394724599501


True

Since p_value(0.47223) > alpha(0.05) 
Inference : Accept H0(null hypothesis) and reject H1(alternative hypothesis)
i.e there is No difference in the diameter of the cutlet between two units

# Problem Statment : 2

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

Minitab File: LabTAT.mtw

HO: No difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list
H1: There is atleast one difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list

As there are multiple smaples

# Perform Anova Test 

In [111]:
df_labTAT = pd.read_csv('/Users/Akshay/Documents/DataScience/ExcelR/Assignments/DataScience/HypothesisTesting/LabTAT.csv')
df_labTAT.head(5)

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [112]:
print('Data Type in Data Frame ---',df_labTAT.dtypes,'\n' , '\n','Data frame Columns Name --- ',df_labTAT.columns,'\n', 'No of rows in df ---',df_labTAT.shape[0], '\n', 
        'No of columns in df ---',df_labTAT.shape[1]  )

Data Type in Data Frame --- Laboratory 1    float64
Laboratory 2    float64
Laboratory 3    float64
Laboratory 4    float64
dtype: object 
 
 Data frame Columns Name ---  Index(['Laboratory 1', 'Laboratory 2', 'Laboratory 3', 'Laboratory 4'], dtype='object') 
 No of rows in df --- 120 
 No of columns in df --- 4


In [113]:
df_labTAT.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [114]:
f_stats, p_value = stats.f_oneway(df_labTAT['Laboratory 1'],df_labTAT['Laboratory 2'],df_labTAT['Laboratory 3'],df_labTAT['Laboratory 4'])
print('f statistic value -- ',f_stats,'\n','p value -- ',p_value)

#5% significance level
alpha = 0.05

#compare p value with alpha value
p_value > alpha

f statistic value --  118.70421654401437 
 p value --  2.1156708949992414e-57


False

Since p_value(~= 0) < alpha(0.05) 
Inference : Reject H0(null hypothesis) and Accept H1(alternative hypothesis)
i.e there is atleast one difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list

# Problem Statment: 3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

ByerRatio.csv


H0:male-female buyer rations are similar across regions i.e male and female variables are independent.
H1:male-female buyer rations are NOT similar across regions i.e male and female variables are dependent on each other

As there is diffrential categorical data

# Perform ChiSquare Test

In [115]:
df_buyerRatio = pd.read_csv('/Users/Akshay/Documents/DataScience/ExcelR/Assignments/DataScience/HypothesisTesting/BuyerRatio.csv', index_col = False)
df_buyerRatio.head(5)

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [116]:
df_buyerRatio = df_buyerRatio.set_index('Observed Values')

In [117]:
df_buyerRatio.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [118]:
print('Data Type in Data Frame ---',df_buyerRatio.dtypes,'\n' , '\n','Data frame Columns Name --- ',df_buyerRatio.columns,'\n', 'No of rows in df ---',df_buyerRatio.shape[0], '\n', 
        'No of columns in df ---',df_buyerRatio.shape[1]  )

Data Type in Data Frame --- East     int64
West     int64
North    int64
South    int64
dtype: object 
 
 Data frame Columns Name ---  Index(['East', 'West', 'North', 'South'], dtype='object') 
 No of rows in df --- 2 
 No of columns in df --- 4


In [119]:
df_buyerRatio

East  West  North  South
Observed Values                          
Males              50   142    131     70
Females           435  1523   1356    750

In [120]:
test_stats,p_value,dof,expectedFreq =  stats.chi2_contingency(df_buyerRatio)
print('test statistical value -- ',test_stats,'\n','p value -- ',p_value,'\n','Degree of freedom --', dof,'\n','Expected Frequency --',expectedFreq)


test statistical value --  1.595945538661058 
 p value --  0.6603094907091882 
 Degree of freedom -- 3 
 Expected Frequency -- [[  42.76531299  146.81287862  131.11756787   72.30424052]
 [ 442.23468701 1518.18712138 1355.88243213  747.69575948]]


There are two methods to interupt once we get the statistical results

1.Comapare test stats with the critical value
2.Compare p value with the alpha value

In [121]:
#1.Comapare test stats with the critical value with 95% confidence
criticalValue = stats.chi2.ppf(.95,df = dof)
print('criticalValue -- ',criticalValue)
test_stats > criticalValue

criticalValue --  7.814727903251179


False

In [122]:
#2.Compare p value with the alpha value
#Considering 5% significance level
alpha = 0.05

#compare p value with alpha value
p_value > alpha

True

Since p_value(0.6603) > alpha(0.05) 
Inference : Accept H0(null hypothesis) and reject H1(alternative hypothesis) i.e male-female buyer rations are similar across regions i.e male and female variables are independent
or
Since test_stats(1.59594) < criticalValue(7.8147): Accept H0(null hypothesis) and reject H1(alternative hypothesis) with a significant level

# Problem Statment: 4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

CustomerOrderForm.csv

H0: No error in any of the customer order forms during audit from all centers
H1: Atleast one error in any of the ceter where customer order forms which needs to be reworked during processing


# Perform ChiSquare Test

In [123]:
df_custOrd = pd.read_csv('/Users/Akshay/Documents/DataScience/ExcelR/Assignments/DataScience/HypothesisTesting/Costomer+OrderForm.csv')
df_custOrd.head(10)

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free
5   Error Free  Error Free  Error Free  Error Free
6   Error Free   Defective  Error Free  Error Free
7   Error Free  Error Free  Error Free  Error Free
8   Error Free  Error Free  Error Free  Error Free
9   Error Free  Error Free  Error Free  Error Free

In [124]:
print('Data Type in Data Frame ---',df_custOrd.dtypes,'\n' , '\n','Data frame Columns Name --- ',df_custOrd.columns,'\n', 'No of rows in df ---',df_custOrd.shape[0], '\n', 
        'No of columns in df ---',df_custOrd.shape[1]  )

Data Type in Data Frame --- Phillippines    object
Indonesia       object
Malta           object
India           object
dtype: object 
 
 Data frame Columns Name ---  Index(['Phillippines', 'Indonesia', 'Malta', 'India'], dtype='object') 
 No of rows in df --- 300 
 No of columns in df --- 4


In [125]:
df_custOrd.isnull().sum()# no N/A values

Phillippines    0
Indonesia       0
Malta           0
India           0
dtype: int64

In [126]:
df_custOrd.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [127]:
df_custOrd.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [128]:
df_custOrd.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [129]:
df_custOrd.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [130]:
custOrdDataArray = {
    'Phillippines':[df_custOrd.Phillippines.value_counts()['Error Free'],df_custOrd.Phillippines.value_counts()['Defective']],
    'Indonesia':[df_custOrd.Indonesia.value_counts()[0],df_custOrd.Indonesia.value_counts()[1]],
    'Malta':[df_custOrd.Malta.value_counts()[0],df_custOrd.Malta.value_counts()[1]],
    'India':[df_custOrd.India.value_counts()[0],df_custOrd.India.value_counts()[1]]
}
custOrdDataArray

{'Phillippines': [271, 29],
 'Indonesia': [267, 33],
 'Malta': [269, 31],
 'India': [280, 20]}

In [131]:
new_df_custOrd = pd.DataFrame(custOrdDataArray)
new_df_custOrd

Phillippines  Indonesia  Malta  India
0           271        267    269    280
1            29         33     31     20

In [132]:
new_df_custOrd=new_df_custOrd.rename(index = {0:'Error Free',1:'Defective'})
new_df_custOrd

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [133]:
test_stats,p_value,dof,expectedFreq =  stats.chi2_contingency(new_df_custOrd)
print('test statistical value -- ',test_stats,'\n','p value -- ',p_value,'\n','Degree of freedom --', dof,'\n','Expected Frequency --',expectedFreq)


test statistical value --  3.858960685820355 
 p value --  0.2771020991233135 
 Degree of freedom -- 3 
 Expected Frequency -- [[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


There are two methods to interupt once we get the statistical results

1.Comapare test stats with the critical value
2.Compare p value with the alpha value

In [134]:
#1.Comapare test stats with the critical value with 95% confidence
criticalValue = stats.chi2.ppf(.95,df = dof)
print('criticalValue -- ',criticalValue)
test_stats > criticalValue

criticalValue --  7.814727903251179


False

In [135]:
#2.Compare p value with the alpha value
#Considering 5% significance level
alpha = 0.05

#compare p value with alpha value
p_value > alpha

True

Since p_value(0.27710) > alpha(0.05) 
Inference : Accept H0(null hypothesis) and reject H1(alternative hypothesis) i.e No error in any of the customer order forms during audit from all centers
or
Since test_stats(3.85896) < criticalValue(7.8147) with a significant level